### DISCUSSION TITLE
problem with parse_B_ext_grid_function

### DISCUSSION THREAD

**Discussion by Aquios7** [BLAST-WarpX/warpx#4663] [Q&A] [answered]
I have a problem with my b-field expanding throughout the entire gridspace when I have it mapped to a specific area in the simulation.

I am trying to put together a simulation that has a plasma enter a b-field that begins at x=0 and scales up in the x direction to a specified number, and on the 0 step the field is correct. The issue comes in the following steps when the b-field seems to expand out from the zone it is set to in the expression, which doesn't seem correct to me from a physics standpoint.

I've added images of the code that I believe may be the problem and I wanted to know if there is a way to fix this or if this is how the field works and I'll try to adjust around it.

Thank you in advance.

![Screenshot from 2024-02-02 16-08-03](https://github.com/ECP-WarpX/WarpX/assets/22086344/e1a5cb76-0664-4ffa-88ac-2c36f9cc2e55)
![Screenshot from 2024-02-02 16-08-19](https://github.com/ECP-WarpX/WarpX/assets/22086344/d45ceda9-71d0-4ef5-8a75-4dbf95294a96)
![Screenshot from 2024-02-02 16-17-36](https://github.com/ECP-WarpX/WarpX/assets/22086344/9fbaff20-4a7f-4830-941c-8eabd67ac95c)
![Screenshot from 2024-02-02 16-17-43](https://github.com/ECP-WarpX/WarpX/assets/22086344/096d0182-ad79-4835-866f-dd33ef05280e)

**Comment by roelof-groenewald** [comment, marked as answer]
Hi @Aquios7.

If you want to just have a static B-field you can simply apply the B-field to the particles (see the docs [here](https://warpx.readthedocs.io/en/latest/usage/parameters.html#applied-to-particles)) instead of writing the B-field to the grid. The `By_external_grid_function` functionality you are using is supplying initial conditions for the B-field, which is then evolved in time using Maxwell's equations.

**Reply by Aquios7** [reply]
Thank you for explaining why the field was moving over time.

If I apply the b-field to the particles, then I don't believe a high beta plasma would be able to affect the b-field. If there isn't a way for the particles to affect the b-field over time then I may have to change how I'm going about my simulation.
